src/main/java/com/books/controllers/BookController.java

```diff
@@ -84,24 +88,26 @@ public ResponseEntity deleteBook(@PathVariable int id){
     }
 
     /**
-     * Increment the rating of a book.
+     * Increment the rating of a book. Also increments the user's rating of the book.
      * @param id the id of the book
      * @return Status OK if successful.
      */
     @PostMapping("/books/{id}/add_rating")
-    public ResponseEntity incrementRating(@PathVariable int id){
+    public ResponseEntity incrementRating(Principal principal, @PathVariable int id){
         bookService.increaseRating(id);
+        userService.addLikedBookId(principal.getName(), id);
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
     /**
-     * Decrement the rating of a book
+     * Decrement the rating of a book. Also decrements the user's rating of the book.
      * @param id the id of the book
      * @return Status OK if successful.
      */
     @PostMapping("/books/{id}/remove_rating")
-    public ResponseEntity decrementRating(@PathVariable int id){
+    public ResponseEntity decrementRating(Principal principal, @PathVariable int id){
         bookService.decreaseRating(id);
+        userService.removeLikedBookId(principal.getName(), id);
```

Comment:
The book's like and rating should be independent. If user likes a book, this should remain even if he decrease the rating

Reply:
As discussed, likes are now favourites.

Reply:
Using the British form - favourites. May change it to American later - favorites.